gpkit/nomials/variables.py

```diff
@@ -135,47 +134,33 @@ def __new__(cls, shape, *args, **descr):
             elif isinstance(arg, Strings) and "label" not in descr:
                 descr["label"] = arg
 
-        values, value_option = [], None
-        for value_option in ["value", "sp_init"]:
-            if value_option in descr:
-                values = descr.pop(value_option)
-                if "vectorization" in descr:
-                    if shape:
-                        values = np.full(shape, values, "f")
-                    else:
-                        descr["value"] = values
-                        values = []
-                break
-
-        descr.pop("vectorization", None)
-
-        valuetype = ""
-        if len(values):
-            if len(shape) == 1:
-                shape_match = len(values) == shape[0]
-                valuetype = "list"
-            else:
-                values = np.array(values)    # pylint: disable=redefined-variable-type
-                shape_match = values.shape == shape
-                valuetype = "array"
-            if not shape_match:
-                raise ValueError("the value's shape must be the same"
-                                 " as the vector's.")
-
         if "name" not in descr:
             descr["name"] = "\\fbox{%s}" % VarKey.new_unnamed_id()
 
+        value_option = None
+        if "value" in descr:
+            value_option = "value"
+        elif "sp_init" in descr:
+            value_option = "sp_init"
```

Comment:
above 5 lines could be `value_option = descr.get("value", descr.get("sp_init", None))`